packages/spec/src/spec.emu.html

```diff
@@ -480,6 +482,26 @@ <h1>Syntactic Grammar</h1>
 AugmentDecoratorArguments :
   `(` ExpressionList `)`
 
+
+DecoratorDeclarationStatement :
+  DecoratorModifiers? `dec` `(` FunctionParameterList? `)`
+
+DecoratorModifiers:
+  `extern`
+
+FunctionDeclarationStatement :
+  FunctionModifiers? `fn` `(` FunctionParameterList? `)` `:` Expression
+
+FunctionModifiers:
+  `extern`
+
+FunctionParameterList : 
+  FunctionParameter
+  FunctionParameterList `,` FunctionParameter
+
+FunctionParameter :
+  `...`? Identifier `?`? `:` Expression
```

Comment:
Do we want the type required for these forms? We don't require the type for template parameters and making it optional saves having to type Cadl.Reflection.Type everywhere I suppose.

Reply:
I guess that can be optional, was more if you made it to the point where you want to declare the parameters for a decorator or function you should probably just do it fully but I guess when we provide in cadl decorator we might not want to restrict that.